lib/provider/ui/index.tsx

```diff
@@ -43,17 +43,17 @@ const Main = styled.main`
 `;
 
 export function UI({ children }: PropsWithChildren): ReactNode {
+  const [theme] = useAtom(themeAtom);
+
   return (
-    <ThemeProvider theme={originalTheme}>
+    <ThemeProvider theme={theme.value}>
       <Style />
       <DirectionProvider dir={i18n.dir}>
-        <Provider>
-          <Header />
-          <Main>
-            {children}
-            <Windows />
-          </Main>
-        </Provider>
+        <Header />
+        <Main>
+          {children}
+          <Windows />
+        </Main>
       </DirectionProvider>
     </ThemeProvider>
```

Comment:
Please keep `Provider`, otherwise we'll have hydration & security issues because all clients share the same `Provider`. You'll want to wrap the provider around this UI component so that useAtom will work.